scripts/build/build/targets.py

```diff
@@ -467,32 +468,37 @@ def BuildASRTarget():
     return target
 
 
-def BuildK32WTarget():
-    target = BuildTarget('k32w', K32WBuilder)
+def BuildNxpTarget():
+    target = BuildTarget('nxp', NxpBuilder)
+    parts = NxpParts.generate_parts()
 
-    # boards
-    target.AppendFixedTargets([
-        TargetPart('k32w0', board=K32WBoard.K32W0),
-        TargetPart('k32w1', board=K32WBoard.K32W1)
-    ])
+    boards = list()
+    for board in parts.boards:
```

Comment:
This seems a bit complex to add in targets.py. 

Any chance you could somehow place it in the nxp pats?

You seem to need some `GetBoardTargets` call.

Also as ugly as this sounds, I would very mych prefer to explicitly add boards without this dictionary complexity. This should read like a config file. I expect you to have maximum 10 boards ... and if so listing them one by one instead of a 3-level nested for and conditionals is better. Do not try to generalize this.

Reply:
Looking at the text file, we basically have k32w0 and k32w1 still. This does not justrify the kwargs and var_modifiers and all that complexity.

Reply:
There are multiple upcoming platforms. I just wanted to use a data-driven approach to avoid people modifying the same file and have them define a JSON instead. The JSON files were also meant to be used by other tools.

Honestly, I thought it would be cleaner to avoid having 50 lines of `target.AppendModifier` calls with different parameters that imply mapping them in the builder constructor, but I guess I was wrong.